library/include/rocblas-functions.h

```diff
@@ -9498,6 +9834,432 @@ ROCBLAS_EXPORT rocblas_status rocblas_zsyrk_strided_batched(rocblas_handle
                                                             rocblas_stride                stride_C,
                                                             rocblas_int batch_count);
 
+/*! \brief BLAS Level 3 API
+
+    \details
+
+    syr2k performs one of the matrix-matrix operations for a symmetric rank-2k update
+
+    C := alpha*(op( A )*op( B )^T + op( B )*op( A )^T) + beta*C
+
+    where  alpha and beta are scalars, op(A) and op(B) are n by k matrix, and
+    C is a symmetric n x n matrix stored as either upper or lower.
+
+        op( A ) = A, op( B ) = B, and A and B are n by k if trans == rocblas_operation_none
+        op( A ) = A^T, op( B ) = B^T,  and A and B are k by n if trans == rocblas_operation_transpose
+
+    @param[in]
+    handle    [rocblas_handle]
+              handle to the rocblas library context queue.
+
+    @param[in]
+    uplo    [rocblas_fill]
+            rocblas_fill_upper:  C is an upper triangular matrix
+            rocblas_fill_lower:  C is a  lower triangular matrix
+
+    @param[in]
+    trans  [rocblas_operation]
```

Comment:
Oddly enough, in the ssyr2k documentation on netlib, they allow a conjugate-transpose operation, presumably this does the same as the simple transpose operation. For the complex cases (csyr2k, eg.) they do not support conjugate-transpose. Seems a little odd, not sure if we should support that as well.

Reply:
Yes that is in the documentation but the code only does simple transpose and non-transpose.  As the other implementations don't mention it (e.g magmaTrans & magmaNoTrans) I think it is better to keep it clear this doesn't do conjugate transpose as Hermitian function her2k handles this.